projects/packages/my-jetpack/_inc/components/plans-section/index.tsx

```diff
@@ -56,31 +64,46 @@ const PlanExpiry: FC< PlanSectionProps > = ( { purchase } ) => {
 	} );
 
 	const expiryMessage = useCallback( () => {
-		const displayDate = dateI18n( 'F jS, Y', expiry_date );
+		const hundredYearDate = getDate( subscribed_date );
+		hundredYearDate.setFullYear( hundredYearDate.getFullYear() + 100 );
+
+		// If expiry_date is null, we'll default to 100 years in the future (same behavior in Store Admin).
+		const expiryDate = expiry_date
+			? dateI18n( 'F jS, Y', expiry_date )
+			: dateI18n( 'F jS, Y', hundredYearDate );
```

Comment:
Could this be shortened like this?
```suggestion
		const expiryDate = dateI18n( 'F jS, Y', expiry_date ?? hundredYearDate );
```